libcfnet/stat_cache.c

```diff
@@ -147,9 +148,17 @@ int cf_remote_stat(AgentConnection *conn, bool encrypt, const char *file,
                  (intmax_t) tloc, file);
         int cipherlen = EncryptString(out, sizeof(out), in, strlen(in) + 1,
                                       conn->encryption_type, conn->session_key);
+
+        tosend = cipherlen + CF_PROTO_OFFSET;
+
+        if(tosend > sizeof(sendbuffer))
+        {
+            ProgrammingError("cf_remote_stat: tosend (%d) > sendbuffer (%ld)",
+                             tosend, sizeof(sendbuffer));
```

Comment:
`ProgrammingError` should be unreachable code. `abort()`ing and generating a core dump is not acceptable behaviour. Except if it's something we didn't think about, and once it happens, we *must* fix it. This is the meaning of `ProgrammingError`. 

`FatalError()` is when you want to exit with error. Still I would return an error to the caller and handle it. Is it hard in this case?